### PR TITLE
Feature test- testing Math-magicians website

### PR DESCRIPTION
**Pull request by ajkacca457**
In this branch I have:

- Refactored calculator functionalities with hooks.
- used router to create routes for homepage, calculator, and quote.
- Styled the website according to the requirement. 
- Made sure that Components are rendered correctly using Jest Snapshots.
- Added test cases to Home, App, and Quote plus the rest of the components.
- Added unit tests for the calculate module and all possible combinations.

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hello @ajkacca457 👋

**Good job so far on this. You did well testing all modules with snapshots. 👍**

However, let's make a couple of fixes and we are good to go. 👌

- Please refer to comments on the code.

### **STATUS: CHANGES REQUESTED** 👷‍♂️

Kindly fix these issues and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

**HAPPY CODING 🎉💻🎉**

**Michgolden Ukeje (CR)**

File: src/__test__/ButtonPanel.test.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import '@testing-library/jest-dom/extend-expect';
+import ButtonPanel from '../components/ButtonPanel';
+/*eslint-disable */
+describe('ButtonPanel Component', () => {
+  it('renders correctly', () => {
+    const originalError = console.error;
+    console.error = jest.fn();
+    const btnpaneltree = TestRenderer
+      .create(<ButtonPanel />)
+      .toJSON();
```
Comment (mikenath223) [inline src/__test__/ButtonPanel.test.js:12]: - [ ] The snapshot test on this component is obsolete kindly update it.
- [ ] It is bad practice to disable the entire eslint rules, if all too necessary it's fair to do that for just a subset of rules.

File: src/__test__/calculate.test.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
+import calculate from '../logic/calculate';
+
+describe('Testing cases for calculate.js Module', () => {
+  test('`AC` button will clear operation', () => {
+    /*eslint-disable */
```
Comment (mikenath223) [inline src/__test__/calculate.test.js:5]: - [ ] Avoid disabling entire eslint rule here also. Kindly fix this on other affected modules too.

**Review by Spayco — APPROVED** [review]
# Project Approved ✔️ 

thank you for making the required changes, your project is now approved!

### please don't forget to merge to master, good luck with the next projects!


Best, 
Aziz Mejri 
_contact me on [twitter](https://twitter.com/iSpaYco) or slack (@SpaYco) if you need any help_

**Pull request merged**